vllm/executor/ray_utils.py

```diff
@@ -31,9 +31,17 @@ def get_node_and_gpu_ids(self) -> Tuple[str, List[int]]:
             gpu_ids = ray.get_gpu_ids()
             return node_id, gpu_ids
 
-        def execute_model_spmd(self, execute_model_req: ExecuteModelRequest):
-            """Used only when SPMD worker and compiled DAG are both
-            enabled."""
+        def execute_model_spmd(
+            self, req_or_tuple: Union[ExecuteModelRequest,
+                                      Tuple[ExecuteModelRequest,
+                                            IntermediateTensors]]):
+            """Execute model in SPMD fashion: used only when SPMD worker and
+            compiled DAG are both enabled.
+
+            Args:
+                req_or_tuple: The request to execute the model, or a tuple
```

Comment:
```suggestion
                req_or_tuple: The request to execute the model, or a tuple
```
QQ: can we just have 2 args?

```
execute_model_spmd(req: ExecuteModelRequest, intermediate_tensors: IntermediateTensors = None)
```

Reply:
With tuple this actually simplifies `_compiled_ray_dag()` and makes it easier to understand.